src/main/java/edu/harvard/iq/dataverse/engine/command/impl/GoogleCloudSubmitToArchiveCommand.java

```diff
@@ -0,0 +1,228 @@
+                if (dataset.getLockFor(Reason.finalizePublication) == null) {
+
+                    String spaceName = dataset.getGlobalId().asString().replace(':', '-').replace('/', '-')
+                            .replace('.', '-').toLowerCase();
+
+                    DataCitation dc = new DataCitation(dv);
+                    Map<String, String> metadata = dc.getDataCiteMetadata();
+                    String dataciteXml = DOIDataCiteRegisterService.getMetadataFromDvObject(
+                            dv.getDataset().getGlobalId().asString(), metadata, dv.getDataset());
+                    String blobIdString = null;
+                    MessageDigest messageDigest = MessageDigest.getInstance("MD5");
+                    try (PipedInputStream dataciteIn = new PipedInputStream(); DigestInputStream digestInputStream = new DigestInputStream(dataciteIn, messageDigest)) {
+                        // Add datacite.xml file
+
+                        new Thread(new Runnable() {
+                            public void run() {
+                                try (PipedOutputStream dataciteOut = new PipedOutputStream(dataciteIn)) {
+
+                                    dataciteOut.write(dataciteXml.getBytes(Charset.forName("utf-8")));
+                                    dataciteOut.close();
+                                } catch (Exception e) {
+                                    logger.severe("Error creating datacite.xml: " + e.getMessage());
+                                    // TODO Auto-generated catch block
+                                    e.printStackTrace();
+                                    throw new RuntimeException("Error creating datacite.xml: " + e.getMessage());
+                                }
+                            }
+                        }).start();
+                        //Have seen broken pipe in PostPublishDataset workflow without this delay
+                        int i=0;
+                        while(digestInputStream.available()<=0 && i<100) {
+                            Thread.sleep(10);
+                            i++;
+                        }
+                        Blob dcXml = bucket.create(spaceName + "/datacite.v" + dv.getFriendlyVersionNumber()+".xml", digestInputStream, "text/xml", Bucket.BlobWriteOption.doesNotExist());
+                        String checksum = dcXml.getMd5ToHexString();
+                        logger.fine("Content: datacite.xml added with checksum: " + checksum);
+                        String localchecksum = Hex.encodeHexString(digestInputStream.getMessageDigest().digest());
+                        if (!checksum.equals(localchecksum)) {
+                            logger.severe(checksum + " not equal to " + localchecksum);
+                            return new Failure("Error in transferring DataCite.xml file to GoogleCloud",
+                                    "GoogleCloud Submission Failure: incomplete metadata transfer");
```

Comment:
Do failures like this show up in the UI? If so, do they need to be translatable into languages other than English?

Reply:
They don't appear anywhere except the log AFAIK, even if you run them as part of a post-publish workflow instead of the API. (Since they run asynchronously, the API just returns an OK, it's launched response. 

The /api/datasets/{id}/actions/:publish also runs archiving, synchronously, if configured, and it does have internationalizable success/failure messages.